vncviewer/DesktopWindow.cxx

```diff
@@ -1014,6 +1004,14 @@ Bool eventIsFocusWithSerial(Display *display, XEvent *event, XPointer arg)
 }
 #endif
 
+void DesktopWindow::maybeGrabKeyboard(bool ungrab)
+{
+  if (fullscreenSystemKeys && fullscreen_active())
+    grabKeyboard();
+  else if (ungrab)
+    ungrabKeyboard();
```

Comment:
I'm cautious about moving `ungrabKeyboard()` in here as well. It will now be called a lot more often than it used to be. It also is a bit surprising given the method's name.

Can't we have just `grabKeyboard()` here?